File: core/core-web/src/test/java/ch/difty/scipamato/core/web/paper/NewsletterChangeEventTest.java
package ch.difty.scipamato.core.web.paper;

import static org.assertj.core.api.Assertions.assertThat;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;
import org.apache.wicket.ajax.AjaxRequestTarget;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
class NewsletterChangeEventTest {

    private NewsletterChangeEvent e;

    @Mock
    private AjaxRequestTarget targetMock, targetMock2;

    @Test
    void canRetrieveTarget() {
        e = new NewsletterChangeEvent(targetMock);
        assertThat(e.getTarget()).isEqualTo(targetMock);
    }

    @Test
    void canOverrideTarget() {
        e = new NewsletterChangeEvent(targetMock);
        assertThat(e.getTarget()).isEqualTo(targetMock);
        e.setTarget(targetMock2);
        assertThat(e.getTarget()).isEqualTo(targetMock2);
    }

    @Test
    void equals() {
        EqualsVerifier
            .forClass(NewsletterChangeEvent.class)
            .withRedefinedSuperclass()
            .suppress(Warning.STRICT_INHERITANCE, Warning.NONFINAL_FIELDS)
            .verify();
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/paper/NewsletterChangeEventTest.kt
package ch.difty.scipamato.core.web.paper

import ch.difty.scipamato.common.AjaxRequestTargetSpy
import nl.jqno.equalsverifier.EqualsVerifier
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Test

internal class NewsletterChangeEventTest {

    private val targetMock = AjaxRequestTargetSpy()
    private val e: NewsletterChangeEvent = NewsletterChangeEvent(targetMock)

    @AfterEach
    fun tearDown() {
        targetMock.reset()
    }

    @Test
    fun canRetrieveTarget() {
        e.target shouldBeEqualTo targetMock
    }

    @Test
    fun equals() {
        EqualsVerifier.simple()
            .forClass(NewsletterChangeEvent::class.java)
            .withRedefinedSuperclass()
            .verify()
    }
}
